packages/gateway/src/features/eventNotification/eventNotificationHandler.ts

```diff
@@ -103,3 +177,112 @@ export async function eventNotificationHandler(
     JSON.stringify(req.payload)
   )
 }
+
+export async function validateAddressesOfTask(bundle: fhir.Bundle) {
+  //validate the patient addresses
+  const patientEntries = bundle.entry?.filter(
+    (e) => e.resource?.resourceType === 'Patient'
+  )
+
+  if (!patientEntries || patientEntries.length === 0) {
+    throw BoomErrorWithCustomMessage(
+      `Could not process the Event Notification, as there were no Person found!`
+    )
+  }
+
+  for (const patient of patientEntries) {
+    const addresses = (patient as fhir.Patient).address
+
+    if (addresses) {
+      for (const address of addresses) {
+        await validateLocationLevelsAndCountry(address)
+      }
+    }
+  }
+
+  // validate event encounter
+  const encounter = getResourceByType<fhir.Encounter>(bundle, 'Encounter')
+  if (!encounter) {
+    throw BoomErrorWithCustomMessage('Encounter entry not found!!')
+  }
+
+  const locationId = encounter.location?.[0].location.reference
+  if (!locationId) {
+    throw BoomErrorWithCustomMessage('Encounter location not found! in bundle!')
+  }
+
+  const location = await fetchFromHearth(`/${locationId}`)
+
+  if (!location || !location.type) {
+    throw BoomErrorWithCustomMessage(
+      `Encounter location with id ${locationId} not found!`
+    )
+  }
+
+  if (location.type.coding?.[0]?.code !== Code.HEALTH_FACILITY) {
+    await validateLocationLevelsAndCountry(location.address)
+  }
+}
+
+export function getResourceByType<T = fhir.Resource>(
+  bundle: fhir.Bundle,
+  type: string
+): T | undefined {
+  const bundleEntry =
+    bundle &&
+    bundle.entry &&
+    bundle.entry.find((entry) => {
+      if (!entry.resource) {
+        return false
+      } else {
+        return entry.resource.resourceType === type
+      }
+    })
+  return bundleEntry && (bundleEntry.resource as T)
+}
+
+export async function validateLocationLevelsAndCountry(address: fhir.Address) {
+  const isCountryValid =
+    address.country === 'FAR' ||
+    (address?.country && lookup.byIso(address?.country))
+
+  if (!isCountryValid) {
+    throw BoomErrorWithCustomMessage(
+      `Could not process the Event Notification, as the supplied country code ${address.country} was incorrect`
+    )
+  }
+
+  if (address.country === DEFAULT_COUNTRY) {
```

Comment:
@Zangetsu101 @naftis we need to find a way to be able to configure FHIR Address Lines array lines to supported location levels for ALL countries so that the Event Notification API can validate any country's administrative structure.  I suggest that there is another prop in the config database used for this or alternatively, if the form.json exists in Gateway as a global constant then, could it be used to validate the structure here.  This is an improvement to this that we should do as part of the Event Location API Validation Epic.

Reply:
I made this ticket: https://github.com/opencrvs/opencrvs-core/issues/5623